src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/SystemDiagnosticsMetricsTrigger/SystemDiagnosticsMetricsTriggerImpl.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.Shared;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.SystemDiagnosticsMetrics
+{
+    // The core implementation of the SystemDiagnosticsMetrics trigger that processes
+    // the trigger settings and evaluates the counter payload. Primary motivation
+    // for the implementation is for unit testability separate from TraceEvent.
+    internal sealed class SystemDiagnosticsMetricsTriggerImpl
+    {
+        private readonly long _intervalTicks;
```

Comment:
Not something we have to do now, but it seems like System.DiagnosticsMetricsTrigger could potentially extend EventCounterTriggerImpl. It's the same functionality (and fields), but adds additional functionality for handling histograms.